clusterloader2/pkg/measurement/util/runtimeobjects/replicaswatcher.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtimeobjects
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+)
+
+const (
+	informerSyncTimeout = time.Minute
+)
+
+// ReplicasWatcher is a struct that allows to check a number of replicas at a given time.
+// Usage:
+// var rw ReplicasWatcher = (...)
+// if err := rw.Start(stopCh); err != nil {
+//   panic(err);
+// }
+// // Get number of replicas as needed.
+// val = rw.Replicas()
+// ...
+// val = rw.Replicas()
+type ReplicasWatcher interface {
+	Replicas() int
+	// Start must block until Replicas() returns a correect value.
+	Start(stopCh chan struct{}) error
+}
+
+// ConstReplicas is a ReplicasWatcher implementation that returns a constant value.
+type ConstReplicas struct {
+	Const int
```

Comment:
I think "Replicas int" would be more intuitive :)

Reply:
I tried that initially, but I got "field and method with the same name Replicas (see details)" compiler error :)

Reply:
Done